tests/core/data_layer/test_data_store.py

```diff
@@ -405,10 +406,10 @@ async def test_batch_update(data_store: DataStore, tree_id: bytes32, use_optimiz
                         key=key, value=value, tree_id=tree_id, use_optimized=False, status=Status.COMMITTED
                     )
                 batch.append({"action": "insert", "key": key, "value": value})
-                keys.append(key)
-            else:
-                key = random.choice(keys)
-                keys.remove(key)
+                keys_values[key] = value
+            elif op_type == 0:
```

Comment:
This modified test should act as a good stress test - we do 60% of operations inserts, 20% upserts and 20% deletes. The whole test runs 1000 operations.